### PR TITLE
Confusing message "unavailable for your platform" (bug 933535)

### PR DESCRIPTION
**Pull request by atiqueahmedziad**

### CODE REVIEW

File: src/media/js/apps.js [hunk 1]
```diff
@@ -113,10 +113,59 @@ define('apps',
...
+            (_.contains(product.device_types, 'android-mobile') ||
+            _.contains(product.device_types, 'android-tablet'))) {
+            reasons.push(gettext('try this app on Android'));
+        }
+        if (!capabilities.firefoxOS &&
+            !_.contains(product.device_types, 'desktop') &&
+            !_.contains(product.device_types, 'android-mobile') &&
+            !_.contains(product.device_types, 'android-tablet') &&
+            _.contains(product.device_types, 'firefoxos')) {
+            reasons.push(gettext('try this app on Firefox OS'));
```
Comment (diox) [inline src/media/js/apps.js]: So, we can simplify these checks. 
1) We don't care about showing incompatibility message for Websites, IMHO.
2) We only need to keep 4 messages, we don't need to show messages like "try this app on Desktop / Android", only one suggestion for each device should be enough.

I suggest to replace all your code by the following : 
```
/* 
 * Return message to show the client when a product is not compatible, trying
 * to find a compatible device to give the user an actionable suggestion.
 */
function getIncompatibilityMessage(product) {
    var message;

    if (_.contains(product.device_types, 'firefoxos')) {
        message = gettext('try this app on Firefox OS'));
    } else if (_.contains(product.device_types, 'android-mobile')) {
        message = gettext('try this app on Firefox for Android Mobile'));
    } else if (_.contains(product.device_types, 'android-tablet')) {
        message = gettext('try this app on Firefox for Android Tablet'));
    } else if (_.contains(product.device_types, 'desktop')) {
        message = gettext('try this app on Firefox'));
    }
    return message;
}

if (!product.isWebsite) {
    // The client is missing some capabilities.
    if (!capabilities.webApps || 
        (!capabilities.packagedWebApps && product.is_packaged)) {
            reasons.push(getIncompatibilityMessage(product));
    } 
    // The product is incompatible with our client.
    else if (!_.contains(product.device_types, device)) {
        reasons.push(getIncompatibilityMessage(product));
    }
}
```

Can you make that work, adjust the unit tests ? Then we'll look at the slimerjs failures.

Comment (diox) [inline src/media/js/apps.js]: You can even simplify my code more by merging the 2 last `if`, since they do the same thing.

Comment (ddurst) [inline src/media/js/apps.js]: Yeah, websites should work everywhere. Having to try them on different platforms is not something we want to track or mitigate in Marketplace (and is a critical issue with showing websites at all).

For the messaging, how do you decide which platform to suggest, in the case of the "Desktop / Android" kinds of things?

Comment (atiqueahmedziad) [inline src/media/js/apps.js]: maybe for website these messages wont show as I hide them in pr #1472

Comment (ddurst) [inline src/media/js/apps.js]: @diox ^ ?

Comment (atiqueahmedziad) [inline src/media/js/apps.js]: hey @ddurst & @diox
@diox thanks for simplifying it :) 
I tried to simplify it as much I can, but can't ignore these "desktop/android", "android/firefox os", "desktop/firefox os". But this has been simplified than in the commit in this pr. So, here, is what I want to write- 
```
/* 
 * Return message to show the client when a product is not compatible, trying
 * to find a compatible device to give the user an actionable suggestion.
 */
function getIncompatibilityMessage(product) {
    var message;

    if (_.contains(product.device_types, 'firefoxos') && 
       (_.contains(product.device_types, 'android-tablet') || 
        _.contains(product.device_types, 'android-mobile')) && 
        _.contains(product.device_types, 'desktop')) {
        message = gettext('try this app on Firefox');
    } else if (_.contains(product.device_types, 'firefoxos') && 
                 (_.contains(product.device_types, 'android-tablet') || 
                  _.contains(product.device_types, 'android-mobile'))) {
        message = gettext('available on Android & Firefox OS');
    } else if ((_.contains(product.device_types, 'android-tablet') || 
                  _.contains(product.device_types, 'android-mobile')) && 
                 _.contains(product.device_types, 'desktop')) {
        message = gettext('available on Desktop & Andorid');
    } else if (_.contains(product.device_types, 'firefoxos') && 
                 _.contains(product.device_types, 'desktop')) {
        message = gettext('available on Firefox Desktop & Firefox OS');
    } else if (_.contains(product.device_types, 'firefoxos')) {
        message = gettext('try this app on Firefox OS');
    } else if (_.contains(product.device_types, 'android-mobile')) {
        message = gettext('try this app on Firefox for Android Mobile');
    } else if (_.contains(product.device_types, 'android-tablet')) {
        message = gettext('try this app on Firefox for Android Tablet');
    } else if (_.contains(product.device_types, 'desktop')) {
        message = gettext('try this app on Firefox Desktop');
    }
    return message;
}

if (!product.isWebsite) {
    // The client is missing some capabilities.
    if (!capabilities.webApps || 
        (!capabilities.packagedWebApps && product.is_packaged) || 
         !_.contains(product.device_types, device)) {
        reasons.push(getIncompatibilityMessage(product));
    };
}
```
Please check if it is enough simplified :)

**Pull request closed without merging**